pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -956,6 +958,45 @@ def decorator(
             self._system_prompt_functions.append(_system_prompt.SystemPromptRunner[AgentDepsT](func, dynamic=dynamic))
             return func
 
+    def output_json_schema(self, output_type: OutputSpec[OutputDataT | RunOutputDataT] | None = None) -> JsonSchema:
+        """The output return JSON schema."""
+        if output_type is None:
+            output_type = self.output_type
+
+        # forces output_type to an iterable
```

Comment:
Let's move all of the logic to a new helper function in `_output.py`, as it only uses `_flatten_output_spec` and types from that module (like `NativeOutput` etc). The new helper function does not need an underscore prefix as the module is already private, and that way we can import it here without needing `# pyright: ignore[reportPrivateUsage]`